Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -239,25 +239,25 @@ void WiiSocket::Update(bool read, bool write, bool except)
       }
       case IOCTL_SO_ACCEPT:
       {
+        s32 ret;
         if (ioctl.buffer_out_size > 0)
         {
           sockaddr_in local_name;
           WiiSockAddrIn* wii_name = (WiiSockAddrIn*)Memory::GetPointer(ioctl.buffer_out);
           WiiSockMan::Convert(*wii_name, local_name);
 
           socklen_t addrlen = sizeof(sockaddr_in);
-          int ret = (s32)accept(fd, (sockaddr*)&local_name, &addrlen);
-          ReturnValue = WiiSockMan::GetNetErrorCode(ret, "SO_ACCEPT", true);
+          ret = static_cast<s32>(accept(fd, (sockaddr*)&local_name, &addrlen));
 
           WiiSockMan::Convert(local_name, *wii_name, addrlen);
         }
         else
         {
-          int ret = (s32)accept(fd, nullptr, nullptr);
-          ReturnValue = WiiSockMan::GetNetErrorCode(ret, "SO_ACCEPT", true);
+          ret = static_cast<s32>(accept(fd, nullptr, nullptr));
         }
 
-        WiiSockMan::GetInstance().AddSocket(ReturnValue);
+        ret = WiiSockMan::GetInstance().AddSocket(ret);
```

Comment:
The return value should be checked for -1/error before being added to the socket table.